core/src/com/unciv/models/ruleset/nation/Personality.kt

```diff
@@ -0,0 +1,91 @@
+package com.unciv.models.ruleset.nation
+
+class Personality {
+
+    var diplomacy:HashMap<String, Float>
+    //Those two are not yet implemented
+    var buildings:HashMap<String, Float>
+    var citystates:HashMap<String, Float>
+
+    constructor(){
+        diplomacy = baseDiplomacy()
+        buildings = baseBuildings()
+        citystates = baseCityStates()
+    }
+    private fun baseDiplomacy():HashMap<String, Float>{
+        return hashMapOf(
+            "war" to 1f,
+            "trade" to 1f,
+            "openborders" to 1f,
+            "friendship" to 1f,
+            "alliance" to 1f
+        )
+    }
+
+    private fun baseBuildings():HashMap<String, Float>{
+        return hashMapOf(
+            "science" to 1f,
+            "production" to 1f,
+            "gold" to 1f,
+            "culture" to 1f,
+            "faith" to 1f,
+            "food" to 1f,
+            "happiness" to 1f,
+            "training" to 1f,
+            "military" to 1f,
+            "wonder" to 1f,
+            "worker" to 1f,
+            "workboat" to 1f
+        )
+    }
+
+    private fun baseCityStates():HashMap<String, Float>{
+        return hashMapOf(
+            "protect" to 1f,
+            "conquer" to 1f,
+            "ally" to 1f,
+            "bully" to 1f,
+            "interact" to 1f
+        )
+    }
+    fun getMinimunMotivationToAttack():Float{
+        //The standard minimun is about 20% of the times motivationToDeclareWar is run
+        var bias = 100 - diplomacy["war"]!! * 20
+        return when {
+            //The attack function behaves diffently across its domain
+            bias >= 89f ->(7.97 * bias - 686.33).toFloat()
+            bias == 80f-> 21f //Standard
+            bias >=78f->(2.77 * bias - 199.47).toFloat()
+            bias >=22f-> (0.59 * bias - 31).toFloat()
+            bias >=11f -> (1.13 * bias - 47).toFloat()
+            bias > 0f -> (2.76 * bias - 65).toFloat()
+            else -> -500f
```

Comment:
What are these magic numbers about...?

Reply:
They are the functions best describing how motivationToDeclareWar behaves, to find the best minimum motivation for a given bias. Could be simplified but results would be less accurate. Plot of motivationToDeclareWar results is below
![WarPlot](https://github.com/yairm210/Unciv/assets/117694699/ef1f63be-5fd2-402c-80a9-1b6803466982)

Reply:
Plot according to what? Where does the blue line come from?